src/advertisingDepartment.ts

```diff
@@ -0,0 +1,23 @@
+import { Client } from './zoo';
+
+// Клас AdvertisingDepartment для управління розсилками і промоціями
+class AdvertisingDepartment {
+    constructor(private clients: Client[]) {}
+
+    // Метод для відправки новинної розсилки клієнтам
+    sendNewsletter(content: string): void {
+        this.clients.forEach(client => {
+            console.log(`Sending newsletter to ${client.name}: ${content}`);
+        });
+    }
+
+    // Метод для відправки промоційної розсилки клієнтам
+    sendPromotions(content: string): void {
```

Comment:
Let's breathe life into our notification system by implementing the Observer pattern across all notification features! Let's make sure that everything that needs to be notified gets the message loud and clear.